src/main/java/com/amihaiemil/eoyaml/AllYamlLines.java

```diff
@@ -45,6 +47,18 @@
  */
 final class AllYamlLines implements YamlLines {
 
+    /**
+     * Pattern to match a sequence or map.
+     */
+    private static final Pattern SEQUENCE_OR_MAP = Pattern.compile("^("
+            + "([ ]*([\\-]|[\\-][ ]+.*))|"
+            + "([ ]*"
+                + "((\"(?:[^\"\\\\]|\\\\.)*\")|"
+                + "('(?:[^'\\\\]|\\\\.)*')|"
+                + "([^\"']*))"
+            +"(:|:[ ].*))"
+            + ")$");
```

Comment:
@newmana Well, almost nobody on Earth will understand this RegEx, so please explain it thoroughly in the JavaDoc above it.

Reply:
Hopefully I've improved this - as much as you can do with regexes.